es-runtime/vite.config.js

```diff
@@ -20,12 +57,18 @@ export default defineConfig({
     base: urlPath,
     outDir: "build",
     minify: false,
+    lib: {
+      name: "es-runtime",
+      entry: listTsFiles('./src'),
```

Comment:
This change was added in response to a vite error ("__vitePreload already declared..."). (When vite tries to add a import of __vitePreload in ExtensionController.js, vite imports the preload-helper.js which defines the __vitePreload into the index.html prior to attempting to load the ExtensionController.js because of this it has technically been already declared resulting in the error.) This has been noted in in the following github issue: https://github.com/vitejs/vite/issues/4016
A weak attempt to resolve it was made in the following pull request: https://github.com/vitejs/vite/pull/4041/files where it simply looks for a "const __vitePreload =" and if it doesn't find it and is not in server side render mode it will add the import anyway. Which doesn't account for other ways of declaring it as vite has done in this specific case.
taking a look at line 274 of the latest source you can see there was a more recent change where "ssr" was replaced with "insertPreload" https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/importAnalysisBuild.ts#L274
InsertPreload is defined as "const insertPreload = !(ssr || !!config.build.lib || isWorker)" as seen on 161 https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/importAnalysisBuild.ts#L161
this means disabling this import no longer relies on finding a const __vitePreload or a server side render flag, it means it can be disabled by having a lib attribute in the build section of the vite config.
by adding the the lib section and setting all the ts files as entry points the issue was resolved